core/arch/arm/mm/tee_mmu.c

```diff
@@ -74,84 +66,88 @@
 /* Support for 31 concurrent sessions */
 static uint32_t g_asid = 0xffffffff;
 
-static void tee_mmu_umap_set_pa(struct tee_mmap_region *tbl,
-			size_t granule, paddr_t pa, size_t size, uint32_t attr)
-{
-	paddr_t upa = ROUNDDOWN(pa, granule);
-	size_t usz = ROUNDUP(pa - upa + size, granule);
-
-	tbl->pa = upa;
-	tbl->size = usz;
-	tbl->attr = attr;
-}
-
-static TEE_Result tee_mmu_umap_add_param(struct tee_mmu_info *mmu, paddr_t pa,
-			size_t size, uint32_t attr)
+static TEE_Result tee_mmu_umap_add_param(struct tee_mmu_info *mmu,
+					 struct param_mem *mem)
 {
-	struct tee_mmap_region *last_entry = NULL;
+	TEE_Result res;
+	struct tee_ta_region *last_entry = NULL;
 	size_t n;
-	paddr_t npa;
+	uint32_t attr = TEE_MMU_UDATA_ATTR;
 	size_t nsz;
+	size_t noffs;
+
+	if (!mobj_is_paged(mem->mobj)) {
+		uint32_t cattr;
+
+		res = mobj_get_cattr(mem->mobj, &cattr);
+		if (res != TEE_SUCCESS)
+			return res;
+		attr |= cattr << TEE_MATTR_CACHE_SHIFT;
+	}
+
+	if (!mobj_is_secure(mem->mobj))
```

Comment:
ok to keep only one of the `mobj_is_nonsec()` or `mobj_is_sec()` as long as any `mobj` creation integrates the required verifications that target memory refernece is either a valid "nonsecure shm" (REE/TEE shm) or a valid "secure shm" (either TA->TA sec shm or SDP sec shm).

Note: if current `mobj_sec_shm` moves to `mobj_is_seccpy_shm`, i think it will be more simple to provide `mobj_is_nonsec(struct mobj *)`, `true` relating to REE/TEE shm, `false` relating to some secure domain shm `(seccpy` shm, or SDP shm, or any other fancy secure domain shm).